response_operations_ui/contexts/reporting_units.py

```diff
@@ -0,0 +1,199 @@
+from flask import url_for
+
+
+def build_reporting_units_context(
+    collection_exercises: list, ru: dict, survey: dict, respondents: list, case, unused_iac: str, permissions: list
+):
+    collection_exercise_section = _build_collection_exercise_section(collection_exercises, ru, survey, permissions)
+    respondents_section = _build_respondents_section(
+        respondents, case, collection_exercises, ru, survey, unused_iac, permissions
+    )
+
+    return {"collection_exercise_section": collection_exercise_section, "respondents_section": respondents_section}
+
+
+def _build_collection_exercise_section(collection_exercises: list, ru: dict, survey: dict, permissions: list):
+    table = []
+    for ce in collection_exercises:
+        row = {
+            "status_class": _select_status_class(ce["responseStatus"]),
+            "hyperlink": url_for(
+                "case_bp.get_response_statuses",
+                ru_ref=ru["sampleUnitRef"],
+                survey=survey["shortName"],
+                period=ce["exerciseRef"],
+            ),
+            "hyperlink_text": "Change" if permissions[0] else "View",
+            "period": ce["exerciseRef"],
+            "reporting_unit_name": ce["companyName"],
+            "trading_as": ce["tradingAs"],
+            "region": ce["companyRegion"],
+            "response_status": ce["responseStatus"],
+            "status": _build_ce_status(ru, survey, ce, permissions),
+        }
+        table.append(row)
+
+    return table
+
+
+def _select_status_class(status):
+    if status == "Not started":
+        return "ons-status--info"
+    elif status == "In progress":
+        return "ons-status--pending"
+    elif status == "Completed" or status == "Completed by phone":
+        return "ons-status--success"
+    elif status == "No longer required":
+        return "ons-status--dead"
+    else:
+        return "ons-status--error"
+
+
+def _build_ce_status(ru, survey, ce, permissions):
+    hyperlink = url_for(
+        "case_bp.get_response_statuses",
+        ru_ref=ru["sampleUnitRef"],
+        survey=survey["shortName"],
+        period=ce["exerciseRef"],
+    )
+    hyperlink_text = "Change" if permissions[0] else "View"
+    status_class = _select_status_class(ce["responseStatus"])
+    response_status = ce["responseStatus"]
+    return (
+        '<span class="ons-status '
+        + status_class
+        + '">'
+        + response_status
+        + "</span>"
+        + "&nbsp;   "
+        + '<a href="'
+        + hyperlink
+        + '">'
+        + hyperlink_text
+        + "</a>"
+    )
+
+
+def _build_respondents_section(
+    respondents: list, case, collection_exercises: list, ru: dict, survey: dict, unused_iac: str, permissions: list
```

Comment:
This is another situation where you should pass what it needs, it will then make the assignment cleaner

Reply:
Updated to receive only what it needs